### PR TITLE
Set the "Link" header

### PR DESCRIPTION
**Pull request by annando**
The "Link" header is part of the Mastodon API. It defines the next or previous items.

### CODE REVIEW

File: src/Module/BaseApi.php [hunk 1]
```diff
@@ -160,6 +175,55 @@ public static function getRequest(array $defaults)
...
+
+		$request = self::$request;
+
+		unset($request['min_id']);
+		unset($request['max_id']);
+		unset($request['since_id']);
+
+		$prev_request = $next_request = $request;
+
+		$prev_request['min_id'] = self::$boundaries['max'] + 1;
```
Comment (MrPetovan) [inline src/Module/BaseApi.php]: Have you verified no item is skipped with this +1? It wasn't part of my original implementation for the follow request list.

Comment (annando) [reply src/Module/BaseApi.php]: The minimum and maximum values for the current request are defined in the `$boundaries` array. So the next or previous items have to be one item before or after, shouldn't they?

Edit: I just checked with "Tooot" from which I know that it uses this header: The results looked really fine.

Comment (MrPetovan) [reply src/Module/BaseApi.php]: How would you know if you skipped one item? Without knowing if the comparison is `id > min_id` or `id >= min_id` we have to test manually by altering the parameter ourselves.

Comment (MrPetovan) [reply src/Module/BaseApi.php]: It looks like they are doing `id > min_id`, so the `+ 1` is unwarranted.

https://github.com/tootsuite/mastodon/blob/c3aef491d66aec743a3a53e934a494f653745b61/app/models/concerns/paginable.rb#L19
```suggestion
		$prev_request['min_id'] = self::$boundaries['max'];
```

File: src/Module/BaseApi.php [hunk 2]
```diff
@@ -160,6 +175,55 @@ public static function getRequest(array $defaults)
...
+		$next_request['max_id'] = self::$boundaries['min'] - 1;
```
Comment (MrPetovan) [inline src/Module/BaseApi.php]: Mastodon is performing `id < max_id` so the + 1 is unwarranted. https://github.com/tootsuite/mastodon/blob/c3aef491d66aec743a3a53e934a494f653745b61/app/models/concerns/paginable.rb#L20

```suggestion
		$next_request['max_id'] = self::$boundaries['min'];
```

**Review by MrPetovan — CHANGES_REQUESTED** [review]

File: src/Module/BaseApi.php (hunk 1 above)
Comment (annando) [reply src/Module/BaseApi.php]: Details :grin:

**Review by MrPetovan — APPROVED** [review]

**Pull request merged**